LocalMultiplayerAgent/DeploymentTool/DeploymentScript.cs

```diff
@@ -0,0 +1,496 @@
+                Console.WriteLine("The specified settings are invalid. Please correct them and re-run the agent.");
+                Environment.Exit(1);
+            }
+
+            if (settings.GameCertificateDetails != null)
+            {
+                await CheckAndUploadCertificatesAsync();
+            }
+
+            if (settings.AssetDetails != null)
+            {
+                foreach (var file in settings.AssetDetails)
+                {
+                    await CheckAndUploadAssetFileAsync(file.LocalFilePath);
+                }
+            }
+
+            dynamic createBuild;
+
+            if (settings.RunContainer)
+            {
+                if (Globals.GameServerEnvironment == GameServerEnvironment.Windows)
+                {
+                    CreateBuildWithManagedContainerRequest request = GetManagedContainerRequest();
+                    createBuild = await CreateBuildWithManagedContainerAsync(request);
+                }
+                else
+                {
+                    await CheckAndUploadLinuxContainerImageAsync();
+
+                    CreateBuildWithCustomContainerRequest request = GetCustomContainerRequest();
+                    createBuild = await CreateBuildWithCustomContainerAsync(request);
+                }
+            }
+            else
+            {
+                CreateBuildWithProcessBasedServerRequest request = GetProcessBasedServerRequest();
+                createBuild = await CreateBuildWithProcessBasedServerAsync(request);
+            }
+
+            if (createBuild.Error != null)
+            {
+                Console.WriteLine("Failed to successfully create build");
+                if (createBuild.Error.ErrorMessage != null)
+                {
+                    Console.WriteLine($"{createBuild.Error.ErrorMessage}");
+                }
+                else if (createBuild.Error.ErrorDetails.Any())
+                {
+                    foreach (var error in createBuild.Error.ErrorDetails)
+                    {
+                        foreach (var errorMessage in error.Value)
+                        {
+                            Console.WriteLine($"{errorMessage}");
+                        }
+                    }
+                    Console.WriteLine($"{createBuild.Error.ErrorMessage}");
+                }
+            }
+            else
+            {
+                //call BuildSummariesAPI to confirm complete deployment ??
+                Console.WriteLine("Build creation was successful!");
+            }
+        }
+
+        public async Task<PlayFabResult<PlayFab.AuthenticationModels.GetEntityTokenResponse>> PlayFabAuthentication()
+        {
+            PlayFabSettings.staticSettings.TitleId = settings.TitleId;
+            var authValidation = new PlayFabResult<PlayFab.AuthenticationModels.GetEntityTokenResponse>();
+            string secret = Environment.GetEnvironmentVariable("PF_SECRET", EnvironmentVariableTarget.Process);
```

Comment:
this is general comment, but it's useful to be verbose when doing CLI apps. For example, in this case, I'd use a Console.WriteLine("trying to read PF_SECRET environment variable"). If it fails to read it, Console.WriteLine("Failed to read PF_SECRET, enter developer secret key").
The golden rule of thumb is to never surprise the user :)